test/invariant/handlers/RateSetterHandler.sol

```diff
@@ -1,29 +1,50 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 pragma solidity ^0.8.13;
 
-import { StdUtils } from "forge-std/StdUtils.sol";
+import { HandlerBase, PSM3 } from "test/invariant/handlers/HandlerBase.sol";
 
 import { MockRateProvider } from "test/mocks/MockRateProvider.sol";
 
-contract RateSetterHandler is StdUtils {
+contract RateSetterHandler is HandlerBase {
 
     uint256 public rate;
 
     MockRateProvider public rateProvider;
 
     uint256 public setRateCount;
 
-    constructor(address rateProvider_, uint256 initialRate) {
+    constructor(PSM3 psm_, address rateProvider_, uint256 initialRate) HandlerBase(psm_) {
         rateProvider = MockRateProvider(rateProvider_);
         rate         = initialRate;
     }
 
     function setRate(uint256 rateIncrease) external {
-        // Increase the rate by up to 100%
-        rate += _bound(rateIncrease, 0, 1e27);
+        // 1. Setup and bounds
 
+        // Increase the rate by up to 20%
+        rate += _bound(rateIncrease, 0, 0.2e27);
+
+        // 2. Cache starting state
+        uint256 startingConversion = psm.convertToShares(1e18);
+        uint256 startingValue      = psm.totalAssets();
+
+        // 3. Perform action against protocol
         rateProvider.__setConversionRate(rate);
 
+        // 4. Perform action-specific assertions
+        assertGe(
+            psm.convertToAssetValue(1e18) + 1,
```

Comment:
Should this not be a comparison with `convertToShares` to match the units of `startingConversion`?

Reply:
Updated to use `convertToAssetValue` for both